File: typescript-connector/src/models/PhysicalCharacteristic.ts
// Class from DFC Business Ontology: #PhysicalCharacteristic
import { SemanticObject } from "../core/SemanticObject.js";
import type { Concept } from "./Concept.js";

export interface PhysicalCharacteristicParams {
  physicalCharacteristicOf?: string;
  date?: string;
  description?: string;
  name?: string;
  characteristicOf?: string;
  hasDimension?: string;
  hasPhysicalDimension?: Concept | string;
}
export class PhysicalCharacteristic extends SemanticObject {
  static get SEMANTIC_TYPE(): string {
    return "dfc-b:PhysicalCharacteristic";
  }

  physicalCharacteristicOf?: string;
  date?: string;
  description?: string;
  name?: string;
  characteristicOf?: string;
  hasDimension?: string;
  hasPhysicalDimension?: Concept | string;

  constructor(
    semanticId: string,
    params?: PhysicalCharacteristicParams,
  ) {
    super(semanticId);
    this.physicalCharacteristicOf = params?.physicalCharacteristicOf;
    this.date = params?.date;
    this.description = params?.description;
    this.name = params?.name;
    this.characteristicOf = params?.characteristicOf;
    this.hasDimension = params?.hasDimension;
    this.hasPhysicalDimension = params?.hasPhysicalDimension;
    this.semanticType = PhysicalCharacteristic.SEMANTIC_TYPE;
    this.registerSemanticProperty("dfc-b:physicalCharacteristicOf", () => this.physicalCharacteristicOf);
    this.registerSemanticProperty("dfc-b:date", () => this.date);
    this.registerSemanticProperty("dfc-b:description", () => this.description);
    this.registerSemanticProperty("dfc-b:name", () => this.name);
    this.registerSemanticProperty("dfc-b:characteristicOf", () => this.characteristicOf);
    this.registerSemanticProperty("dfc-b:hasDimension", () => this.hasDimension);
    this.registerSemanticProperty("dfc-b:hasPhysicalDimension", () => this.hasPhysicalDimension);
  }
  static {
    SemanticObject.typeRegistry.set(PhysicalCharacteristic.SEMANTIC_TYPE, PhysicalCharacteristic);
  }
}

File: ruby-gem/lib/models/physical_characteristic.rb
# frozen_string_literal: true

# Class from DFC Business Ontology: #PhysicalCharacteristic
require_relative '../core/semantic_object'


module DfcLinkmlConnector
  module Models
    class PhysicalCharacteristic < Core::SemanticObject
      SEMANTIC_TYPE = "dfc-b:PhysicalCharacteristic".freeze

      # @return [String]
      attr_accessor :physical_characteristic_of

      # @return [String]
      attr_accessor :date

      # @return [String]
      attr_accessor :description

      # @return [String]
      attr_accessor :name

      # @return [String]
      attr_accessor :characteristic_of

      # @return [String]
      attr_accessor :dimension

      # @return [Concept, String]
      attr_accessor :physical_dimension

      # @param semanticId [String]
      # @param physicalCharacteristicOf: nil, date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil, physicalDimension: nil
      def initialize(semanticId, physicalCharacteristicOf: nil, date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil, physicalDimension: nil)
        super(semanticId)
        @physical_characteristic_of = physicalCharacteristicOf
        @date = date
        @description = description
        @name = name
        @characteristic_of = characteristicOf
        @dimension = dimension
        @physical_dimension = physicalDimension
        self.semanticType = "dfc-b:PhysicalCharacteristic"
        registerSemanticProperty("dfc-b:physicalCharacteristicOf", &method("physical_characteristic_of")).valueSetter = method("physical_characteristic_of=")
        registerSemanticProperty("dfc-b:date", &method("date")).valueSetter = method("date=")
        registerSemanticProperty("dfc-b:description", &method("description")).valueSetter = method("description=")
        registerSemanticProperty("dfc-b:name", &method("name")).valueSetter = method("name=")
        registerSemanticProperty("dfc-b:characteristicOf", &method("characteristic_of")).valueSetter = method("characteristic_of=")
        registerSemanticProperty("dfc-b:hasDimension", &method("dimension")).valueSetter = method("dimension=")
        registerSemanticProperty("dfc-b:hasPhysicalDimension", &method("physical_dimension")).valueSetter = method("physical_dimension=")
      end

      Core::SemanticObject.type_registry[SEMANTIC_TYPE] = self
    end
  end
end
